src/typechecker.erl

```diff
@@ -4210,6 +4210,8 @@ refinable(?type(pos_integer), _Env, _Trace) ->
     true;
 refinable(?type(neg_integer), _Env, _Trace) ->
     true;
+refinable(?type(boolean), _Env, _Trace) ->
+    true;
```

Comment:
Isn't refinable/3 always called with a normalized type? (If not, maybe that's a bug somewhere?) `boolean()` is normalized to the union `false | true`, which is refinable already.

Reply:
It's not. It is called only from `check_arg_exhaustiveness/4` (via `all_refinable/2`), and apparently the `ArgTys` there aren't normalised. Also, `refinable/3` has clauses for `char()`, `string()` or `binary()`, and all of these are non-normalised types, so it's probably meant to be called with non-normalised types.

But I was also surprised when I discovered that it's not made for normalised types. Handling only normalised types would make more sense to me (from my limited perspective).